src/ess/isissans/general.py

```diff
@@ -52,45 +74,68 @@ def default_parameters() -> dict:
     }
 
 
-def get_detector_data(
-    dg: LoadedFileContents[RunType],
-    sample_offset: SampleOffset,
-    detector_bank_offset: DetectorBankOffset,
-) -> RawDetector[RunType]:
+def to_detector_position_offset(
+    global_offset: DetectorBankOffset, beam_center: BeamCenter
+) -> DetectorPositionOffset[RunType]:
+    return DetectorPositionOffset[RunType](global_offset - beam_center)
+
+
+def to_monitor_position_offset(
+    global_offset: MonitorOffset[MonitorType],
+) -> MonitorPositionOffset[RunType, MonitorType]:
+    return MonitorPositionOffset[RunType, MonitorType](global_offset)
```

Comment:
These two helpers replace logic in the removed `components.py`, which was mostly duplicating more generic code.